File: app/src/main/java/com/esoxjem/movieguide/listing/MoviesListingView.java
package com.esoxjem.movieguide.listing;

import com.esoxjem.movieguide.Movie;

import java.util.List;

/**
 * @author arun
 */
interface MoviesListingView
{
    void showMovies(List<Movie> movies);
    void loadingStarted();
    void loadingFailed(String errorMessage);
    void onMovieClicked(Movie movie);
}

File: app/src/main/java/com/esoxjem/movieguide/movies/listing/MoviesListingView.kt
package com.esoxjem.movieguide.movies.listing

import com.esoxjem.movieguide.movies.entities.Movie

/**
 * @author arun
 */
interface MoviesListingView {
    fun showMovies(movies: List<Movie>)
    fun loadingStarted()
    fun loadingFailed(errorMessage: String?)
    fun onMovieClicked(movie: Movie)
}
